backend/LeetWars.Builder/Services/ConsumeMessages.cs

```diff
@@ -7,20 +12,40 @@ namespace LeetWars.Builder.Services
     public class ConsumeMessages : BackgroundService
     {
         private readonly IConsumerService _consumerService;
-
-        public ConsumeMessages(IConsumerService consumerService)
+        private readonly ICodeRunManagerService _codeRunManagerService;
+        private readonly IProducerService _producerService;
+        public ConsumeMessages(IConsumerService consumerService, ICodeRunManagerService codeRunManagerService, IProducerService producerService)
         {
             _consumerService = consumerService;
+            _codeRunManagerService = codeRunManagerService;
+            _producerService = producerService;
         }
 
         protected override Task ExecuteAsync(CancellationToken stoppingToken)
         {
-            var handler = new EventHandler<BasicDeliverEventArgs>((model, args) =>
+            var handler = new EventHandler<BasicDeliverEventArgs>(async (model, args) =>
             {
                 var body = args.Body.ToArray();
-                var message = Encoding.UTF8.GetString(body).Replace('"', '\0');
+                var message = Encoding.UTF8.GetString(body);
+
+                var request = JsonConvert.DeserializeObject<CodeRunRequest>(message);
+
+                if (request != null && Models.HelperModels.Languages.AvailableLanguages.Contains(request.Language))
+                {
+                    var result = await _codeRunManagerService.Run(request);
+
+                    var settings = new JsonSerializerSettings
+                    {
+                        ContractResolver = new CamelCasePropertyNamesContractResolver()
+                    };
+
+                    var codeRunResultJson = JsonConvert.SerializeObject(result, settings);
+
+                    var notifierMessage = new NotifierMessage() { HubName= "codeDisplayingHub", Message = codeRunResultJson };
```

Comment:
why do you need a hub name here?

Reply:
because we use the same queue and HubName is a parameter that helps determine to which signalR hub I need to pass the message